src/subproblem_callback.jl

```diff
@@ -0,0 +1,57 @@
+"""
+Struct for user supbroblem callbacks.
+
+**Fields**
+
+* `name::String` : User's name for the callback. Defaults to `string(h)`.
+* `h::Function` : Callback function. See notes below for calling signature.
+* `ext::Dict{Symbol,Any}` : Dictionary to store data between callback calls or needed parameters.
+
+The callback function `h` must have the signature
+    `h(ext::Dict{Symbol,Any}, sp::SubproblemRecord, niter::Int, scenario_id::ScenarioID)::Bool`
```

Comment:
A couple of things here.  First, it doesn't seem like the subproblem callback needs to return anything. The return value isn't used at all so you can leave it entirely unspecified and remove one thing for the user to worry about.  For the normal callbacks, the return value allows the user to terminate PH for any reason they see fit.  Here we don't have that need.

Second, the subproblem record is meant to contain things that PH needs to run but that the user doesn't need to know anything about.  That's why they are stored in a separate data struct and not in the subproblem itself (it's a data encapsulation issue).  If the user changes these data fields (on purpose or accidentally), PH could break in very unusual and non-obvious ways.  So the `SubproblemRecord` should not be exposed to the user and only the subproblem should be handed to the callback.

Reply:
Ah ok, I was wondering about that. Makes sense, will fix accordingly.